common/src/main/resources/db/changelog/v2022/create-analyst-view.sql

```diff
@@ -0,0 +1,142 @@
+                       LIMIT 1000
+    )
+    LOOP
+
+        ubc := (SELECT COUNT(*) FROM benes_served);
+RAISE NOTICE 'Inserting benes for %', con_number;
+RAISE NOTICE 'Benes before %', ubc;
+
+        -- Get most recent run pulling data for the contract
+last_run := (
+            SELECT created_at
+            FROM job_view WHERE status = 'SUCCESSFUL' AND contract_number = con_number
+            ORDER BY created_at DESC
+            LIMIT 1
+        );
+RAISE NOTICE 'Last run %', last_run;
+
+        -- Use most recent job to determine what months of enrollment for the contract have been queried.
+        -- Once those months are known, use the temp table to deduplicate the enrollment against other contracts
+INSERT INTO benes_served(beneficiary_id)
+SELECT DISTINCT(beneficiary_id)
+FROM coverage_view
+WHERE contract = con_number AND year IN (2020, 2021, 2022, 2023)
+  AND bene_coverage_search_event_id IN (
+    -- Find all search events with coverage (should be one per month)
+    SELECT bene_coverage_search_event_id
+    FROM (
+             SELECT coverage_view.year, coverage_view.month, coverage_view.bene_coverage_search_event_id
+             FROM coverage_view
+                      INNER JOIN bcp_view ON coverage_view.bene_coverage_period_id = bcp_view.bcp_id
+             WHERE bcp_view.status = 'SUCCESSFUL' AND coverage_view.contract = con_number
+               AND bcp_view.year IN (2020, 2021, 2022, 2023)
+             GROUP BY coverage_view.contract, coverage_view.year, coverage_view.month, coverage_view.bene_coverage_search_event_id
+             ORDER BY coverage_view.contract, coverage_view.year, coverage_view.month
+         ) coverage_meta
+    WHERE make_timestamptz(coverage_meta.year, coverage_meta.month, 1, 0, 0, 0, 'America/New_York') < last_run
+)
+ON CONFLICT (beneficiary_id) DO NOTHING;
+
+END LOOP;
+
+ubc := (SELECT COUNT(*) FROM benes_served);
+
+DROP TABLE IF EXISTS benes_served;
+
+UPDATE ab2d_statistics
+SET statistic_value = ubc
+WHERE statistic_name = 'total_benes_served';
+END
+$$;
+
+------------------------------
+-- Grants to ab2d_analyst user
+------------------------------
+
+CREATE ROLE ab2d_analyst noinherit;
+
+-- Grant read only privileges to a subset of tables in the database
+GRANT SELECT ON event_api_response TO ab2d_analyst;
+GRANT SELECT ON event_api_request TO ab2d_analyst;
+GRANT SELECT ON event_bene_coverage_search_status_change TO ab2d_analyst;
+GRANT SELECT ON event_bene_reload TO ab2d_analyst;
+GRANT SELECT ON event_bene_search TO ab2d_analyst;
+GRANT SELECT ON event_error TO ab2d_analyst;
+GRANT SELECT ON event_file TO ab2d_analyst;
+GRANT SELECT ON event_job_status_change TO ab2d_analyst;
+GRANT SELECT ON job_output TO ab2d_analyst;
+GRANT SELECT ON contract_view TO ab2d_analyst;
+GRANT SELECT ON job_view TO ab2d_analyst;
+GRANT SELECT ON bcp_view TO ab2d_analyst;
+GRANT SELECT ON coverage_view TO ab2d_analyst;
+
+GRANT SELECT, UPDATE, INSERT ON ab2d_statistics TO ab2d_analyst;
```

Comment:
Check with Will but I don't think you want anything to do with the analyst in this checkin.

Reply:
@wnyffenegger Comment?

Reply:
Well I can't control permissions from ab2d-ops if those views don't already exist so I would include everything.

Reply:
We're assuming that the ab2d_analyst is already created in the environment and you're just populating the views it can use. It starts with zero view of anything so you're going to have to do the grants.

Reply:
@sb-dkniesteadt one thing though, get rid of the create role part. Assume that the role already exists in the environment.

Reply:
Tricky, I don't think so. Maybe that changes things. I don't want to deal with that mess. It's a chicken and egg problem because during the deployment of the database these views won't exist. It's only when a worker/api start up that it will.